### ISSUE TITLE
Assemble the whole pipeline

### ISSUE THREAD

**Issue by tschimbr** [eonum/medcodelearn#3]
One should be able to run the following tasks in one single run:
1. Generate tokens based on classifications
2. Generate vectorization based on this tokens and eventually training data.
3. Train and validate (classifier specific evaluation metrics) classifiers using training data. Use the training data only for training and validation. Split the training data into training and evaluation set.
4. Evaluate the whole pipeline with global evaluators (see https://github.com/eonum/medcodelearn/issues/2) using a global test set that has not yet been used in the previous steps.

**Comment by tschimbr** [comment]
And add support for hyper-parameter validation over the whole pipeline.

**Comment by tschimbr** [comment]
Done for Flat classification
